### ISSUE TITLE
src/stub_write.c:2:10: fatal error: 'sys/types.h' file not found

### ISSUE THREAD

**Issue by orj** [hth313/Calypsi-65816-hello-world#2]
1. I've installed v4.4 of Calypsi-65816 on my Mac.
2. I've cloned this hello world repo with recursive submodules.
3. When running `make hello.pgz` I get the following error:

```
src/stub_write.c:2:10: fatal error: 'sys/types.h' file not found
```

There doesn't appear to be a `sys/types.h` in any of the Calypsi include folders.

**Comment by orj** [comment]
I commented out this line and compilation got further but then bailed during linking:

```
directive '.sectionEnd(data_init_table)' referenced from section 'code' at offset 000001 in Foenix-startup-lc-sd.o: value 131081 is out of range, allowed range is -32768 to 65535
directive '.sectionStart(data_init_table)' referenced from section 'code' at offset 000006 in Foenix-startup-lc-sd.o: value 131072 is out of range, allowed range is -32768 to 65535
```

**Comment by orj** [comment]
I tried updating the the submodule to the latest code on `main`.  This got rid of the compiler issues.  Now it complains during linking:
```
module/Calypsi-65816-Foenix/linker-files/Foenix-U.scm: openFile: does not exist (No such file or directory)
```

**Comment by orj** [comment]
I changed line `13` of the `Makefile` to:

```make
FOENIX_LINKER_RULES = $(FOENIX)/linker-files/c256-fmx-plain.scm
```

And now make completes successfully. I have an FMX so that makes sense for me.

**Comment by hth313** [comment]
Yeah, as mentioned in the Foenix discord I cleaned up the C library headers to remove POSIX stuff and as part of that the `sys/types.h` header was removed, as it is POSIX, not standard C.

I am in the progress of including the Foenix 65816 board support with the Calypsi tools for the 5.1 release which should make it easier, and also update the example file. However, I expect this may be late September release.
There is also current work to port the MCP for the FMX/U machines and I may switch to that kernel instead if it becomes available. The current 65816 kernel is block oriented in the I/O which does not match well with the Calypsi C stub interface, the MCP should be a better fit.

So, we have to wait and see with that, but I get the impression that you got something together and that you are not stuck on it at the moment?

**Comment by orj** [comment]
Yeah I did manage to resolve the issues I was having as described above.
